### ISSUE TITLE
Base 10 error when writing to gist file

### ISSUE THREAD

**Issue by Nonesence999** [PyGithub/PyGithub#2704]
```python
from github import Github
import github.InputFileContent
# Authentication is defined via github.Auth
from github import Auth

# using an access token
auth = Auth.Token("TOKEN")

g = Github(auth=auth)

gist = g.get_gist("ID")
# Get file content
file_content = gist.files["100.md"].content

print(file_content) #content is being printed

gist.edit("",{"100.md": github.InputFileContent("hello world")},)
```
The token is right , the id is right , the 100.md file exist 

line 694, in __requestEncode
    int(responseHeaders[Consts.headerRateRemaining]),
ValueError: invalid literal for int() with base 10: '195.0'

**Comment by semick-dev** [comment]
Confirm I'm hitting this as well. Python `3.11.1` - Windows.

**Comment by Nonesence999** [comment]
Apparently its fixed but still not merged after 6+ days

**Comment by EnricoMi** [comment]
Fixed in #2697 and released as v1.59.1. Thanks for reporting!
